### DISCUSSION TITLE
Is there any way to include an external .so file into a program?

### DISCUSSION THREAD

**Discussion by Minater247** [cc-tweaked/CC-Tweaked#1484] [Q&A]
I've been attempting to write a tracker program (...in case anyone doesn't know, a type of 80s music-making program) which plays back audio through the speakers. On paper, everything should work fine, and I am in fact able to play back the audio. Unfortunately, the dynamically allocated Lua tables have too much overhead for sufficient real-time audio processing capabilities, and it takes far too long to calculate the audio buffer for it to actually work.

Now that's a lot of context, but it leads up to the solution - writing some of the code in C. I've done this before, at the guidance of [this post](https://chsasank.com/lua-c-wrapping.html), and I'm 99% sure that this would solve a lot of my performance issues. However, I've found no way to include the type of library Lua can read, a .so file, into a CC:T computer, since require (both ```require ''``` and ```require("")``` syntax) only look for raw-text Lua files.

I'm hoping someone here knows a bit more than I do, then - how can I include one of these library files into a CC:T program?

**Comment by SquidDev** [comment]
I'm afraid this is not possible, for a couple of reasons:
 - CC's Lua VM is written in Java, and so is not compatible with extension modules written targeting the C Lua API. 
 - It's not really possible to sandbox arbitrary object files, and so allowing loading them would be a major security risk.

**Reply by Minater247** [reply]
Ah welp, worth a shot anyways, that makes sense. Thank you very much!
